src/Shared/Modifiers.cs

```diff
@@ -471,19 +471,26 @@ internal static string GetItemSpecModifier(string currentDirectory, string itemS
 
                         if (NativeMethodsShared.IsWindows)
                         {
-                            Match root = FileUtilitiesRegex.DrivePattern.Match(modifiedItemSpec);
-
-                            if (!root.Success)
+                            int length = -1;
+                            if (FileUtilitiesRegex.DrivePattern.IsMatch(modifiedItemSpec))
+                            {
+                                length = 2;
+                            }
+                            else
                             {
-                                root = FileUtilitiesRegex.UNCPattern.Match(modifiedItemSpec);
+                                var match = FileUtilitiesRegex.StartsWithUncPattern.Match(modifiedItemSpec);
+                                if (match.Success)
+                                {
+                                    length = match.Length;
+                                }
                             }
 
-                            if (root.Success)
+                            if (length != -1)
                             {
-                                ErrorUtilities.VerifyThrow((modifiedItemSpec.Length > root.Length) && IsSlash(modifiedItemSpec[root.Length]),
+                                ErrorUtilities.VerifyThrow((modifiedItemSpec.Length > length) && IsSlash(modifiedItemSpec[length]),
```

Comment:
What happens with "C:[space]" have we already trimed this?

Reply:
Say `modifiedItemSpec == "c: "` (with space after colon), then previously the regex would match two characters, so the length check would fail (2 != 3).

Reply:
Okay, so you haven't changed behavior here?

Reply:
If this PR changes behaviour it is a bug.